control-plane/pkg/reconciler/source/v2/sourcev2.go

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sources "knative.dev/eventing-kafka/pkg/apis/sources/v1beta1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/reconciler"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup"
+
+	internals "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing"
+	internalscg "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+	internalsclient "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/clientset/versioned"
+	internalslst "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/listers/eventing/v1alpha1"
+)
+
+const (
+	DefaultDeliveryOrder = internals.Ordered
+)
+
+type ReconcilerV2 struct {
+	ConsumerGroupLister internalslst.ConsumerGroupLister
+	InternalsClient     internalsclient.Interface
+}
+
+func (r *ReconcilerV2) ReconcileKind(ctx context.Context, ks *sources.KafkaSource) reconciler.Event {
+
+	//todo initialize consumer group offsets here?
+
+	consgroup, err := r.reconcileConsumerGroup(ctx, ks)
+	if err != nil {
+		ks.GetConditionSet().Manage(&ks.Status).MarkFalse(consumergroup.KafkaConditionConsumerGroupNotReady, "failed to reconcile consumer group", err.Error())
+		return err
+	}
+	ks.GetConditionSet().Manage(&ks.Status).MarkTrue(consumergroup.KafkaConditionConsumerGroupReady)
```

Comment:
We need to make sure that the condition `KafkaConditionConsumerGroupNotReady` is registered and the call `GetConditionSet()` method on the KafkaSource object returns all conditions that are relevant to this reconciler.

See `RegisterAlternateConditionSet` for the Broker.

Reply:
Sure. Where can I find `RegisterAlternateConditionSet`?

Reply:
Sorry, it's called `RegisterAlternateBrokerConditionSet`

https://github.com/knative/eventing/blob/d98ae8c06a46d930d08d8158a3f2962a99d9efca/pkg/apis/eventing/v1/broker_lifecycle.go#L44